core/src/main/scala/com/yahoo/maha/core/query/oracle/OuterGroupByQueryGenerator.scala

```diff
@@ -0,0 +1,443 @@
+        case _=> throw new IllegalArgumentException(s"Unexpected fact derived column found in outer select $column")
+      }
+    }
+
+    def ogbGenerateOuterColumns(): Unit = {
+      // add requested dim and fact columns, this should include constants
+      queryContext.requestModel.requestCols foreach {
+        columnInfo =>
+          if (!columnInfo.isInstanceOf[ConstantColumnInfo] && queryBuilderContext.aliasColumnMap.contains(columnInfo.alias)) {
+            aliasColumnMapOfRequestCols += (columnInfo.alias -> queryBuilderContext.aliasColumnMap(columnInfo.alias))
+          } else if (queryContext.factBestCandidate.duplicateAliasMapping.contains(columnInfo.alias)) {
+            val sourceAliases = queryContext.factBestCandidate.duplicateAliasMapping(columnInfo.alias)
+            val sourceAlias = sourceAliases.find(queryBuilderContext.aliasColumnMap.contains)
+            require(sourceAlias.isDefined
+              , s"Failed to find source column for duplicate alias mapping : ${queryContext.factBestCandidate.duplicateAliasMapping(columnInfo.alias)}")
+            aliasColumnMapOfRequestCols += (columnInfo.alias -> queryBuilderContext.aliasColumnMap(sourceAlias.get))
+          }
+          val renderedCol = columnInfo match {
+            case FactColumnInfo(alias) if queryBuilderContext.containsPreOuterAlias(alias) =>
+              val preOuterAliasOption = queryBuilderContext.getPreOuterFinalAliasToAliasMap(alias)
+              if(preOuterAliasOption.isDefined) {
+                val preOuterAlias = s""""${preOuterAliasOption.get}""""
+                s"""$preOuterAlias AS "$alias""""
+              } else {
+                s""""$alias"""
+              }
+            case FactColumnInfo(alias) =>
+              val column = queryBuilderContext.getFactColByAlias(alias)
+              renderParentOuterDerivedFactCols(alias, column)
+            case DimColumnInfo(alias) => s""""$alias""""
+            case ConstantColumnInfo(alias, value) =>
+              s"""'$value' AS "$alias""""
+            case _ => throw new UnsupportedOperationException("Unsupported Column Type")
+          }
+          queryBuilder.addOuterColumn(renderedCol)
+      }
+
+      if (queryContext.requestModel.includeRowCount) {
+        queryBuilder.addOuterColumn(OracleQueryGenerator.ROW_COUNT_ALIAS)
+        aliasColumnMapOfRequestCols += (OracleQueryGenerator.ROW_COUNT_ALIAS -> PAGINATION_ROW_COUNT_COL)
+      }
+
+      val outerWhereClause = generateOuterWhereClause(queryContext, queryBuilderContext)
+      queryBuilder.setOuterWhereClause(outerWhereClause.toString)
+
+    }
+
+    def ogbGeneratePreOuterColumns(primitiveInnerAliasColMap: Map[String, Column]): Unit = {
+      // add requested dim and fact columns, this should include constants
+      val preOuterRenderedColAlias = new mutable.HashSet[String]()
+      queryContext.requestModel.requestCols foreach {
+        columnInfo =>
+
+          columnInfo match {
```

Comment:
Handle default case for this pattern match

Reply:
done